common/src/chain/config/regtest.rs

```diff
@@ -0,0 +1,180 @@
+
+        Self {
+            pool_id,
+            stake_private_key,
+            vrf_private_key,
+        }
+    }
+
+    pub fn pool_id(&self) -> PoolId {
+        self.pool_id
+    }
+
+    pub fn stake_private_key(&self) -> PrivateKey {
+        self.stake_private_key.clone()
+    }
+
+    pub fn vrf_private_key(&self) -> VRFPrivateKey {
+        self.vrf_private_key.clone()
+    }
+}
+
+impl Default for GenesisStakingSettings {
+    fn default() -> Self {
+        Self::new("")
+    }
+}
+
+impl FromStr for GenesisStakingSettings {
+    type Err = Infallible;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(Self::new(s))
+    }
+}
+
+impl Display for GenesisStakingSettings {
+    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
+        write!(
+            f,
+            "{{pool_id: {:?}, stake_private_key: {:?}, vrf_private_key: {:?}}}",
+            self.pool_id(),
+            self.stake_private_key(),
+            self.vrf_private_key()
+        )
+    }
+}
+
+pub fn genesis_values(
+    genesis_staking_settings: GenesisStakingSettings,
+) -> (
+    PoolId,
+    Box<StakePoolData>,
+    PrivateKey,
+    PublicKey,
+    VRFPrivateKey,
+    VRFPublicKey,
+) {
+    let genesis_stake_private_key = genesis_staking_settings.stake_private_key();
+    let genesis_stake_public_key = PublicKey::from_private_key(&genesis_stake_private_key);
+
+    let genesis_vrf_private_key = genesis_staking_settings.vrf_private_key();
+    let genesis_vrf_public_key = VRFPublicKey::from_private_key(&genesis_vrf_private_key);
+
+    let genesis_pool_stake_data = Box::new(StakePoolData::new(
+        MIN_STAKE_POOL_PLEDGE,
+        Destination::PublicKey(genesis_stake_public_key.clone()),
+        genesis_vrf_public_key.clone(),
+        Destination::PublicKey(genesis_stake_public_key.clone()),
+        PerThousand::new(1000).expect("Valid per thousand"),
+        Amount::ZERO,
+    ));
+
+    (
+        genesis_staking_settings.pool_id(),
+        genesis_pool_stake_data,
+        genesis_stake_private_key,
+        genesis_stake_public_key,
+        genesis_vrf_private_key,
+        genesis_vrf_public_key,
+    )
+}
+
+pub fn create_regtest_pos_genesis(
+    genesis_staking_settings: GenesisStakingSettings,
+    premine_destination: Destination,
+) -> Genesis {
+    let (
+        genesis_pool_id,
+        genesis_stake_pool_data,
+        _genesis_stake_private_key,
+        _genesis_stake_public_key,
+        _genesis_vrf_private_key,
+        _genesis_vrf_public_key,
+    ) = genesis_values(genesis_staking_settings);
+
+    let create_genesis_pool_txoutput =
+        TxOutput::CreateStakePool(genesis_pool_id, genesis_stake_pool_data);
+
+    let premine_output =
+        TxOutput::Transfer(OutputValue::Coin(DEFAULT_INITIAL_MINT), premine_destination);
+
+    Genesis::new(
+        String::new(),
+        BlockTimestamp::from_int_seconds(1639975460),
+        vec![premine_output, create_genesis_pool_txoutput],
+    )
+}
```

Comment:
Some basic unit tests that explain the parsing would really help.

You don't have to test every corner case (please don't spend days creating them), but usually unit tests can be used by beginners to learn how a struct works.

Reply:
No problem. Added.

`mod.rs`s' `decode_hex()` was breaking these tests because it panicks when decoding fails. Because I didn't want to change the upstream function, I've copied it into `regtest.rs` and made that return `Result`. If you want me to change upstreams to return a `Result` too and remove the duplicated code, let me know.

Reply:
Are you talking about [#1053](https://github.com/mintlayer/mintlayer-core/issues/1053)?

Reply:
No, this was different. For example, in the `invalid_vrf_private_key()` test, because the input wasn't long enough nor hex, `mod.rs`s `decode_hex()` would panic and so my test wouldn't pass.